blueocean-pipeline-api-impl/src/main/java/io/jenkins/blueocean/BlueOceanUrlMapperImpl.java

```diff
@@ -0,0 +1,140 @@
+package io.jenkins.blueocean;
+
+import hudson.Extension;
+import hudson.model.Item;
+import hudson.model.ItemGroup;
+import hudson.model.Job;
+import hudson.model.ModelObject;
+import hudson.model.Run;
+import io.jenkins.blueocean.rest.factory.BlueOceanUrlMapper;
+import io.jenkins.blueocean.rest.factory.BluePipelineFactory;
+import io.jenkins.blueocean.rest.factory.organization.OrganizationFactory;
+import io.jenkins.blueocean.rest.model.BlueMultiBranchPipeline;
+import io.jenkins.blueocean.rest.model.BlueOrganization;
+import io.jenkins.blueocean.rest.model.BluePipeline;
+import io.jenkins.blueocean.rest.model.Resource;
+import jenkins.branch.MultiBranchProject;
+import jenkins.model.ModifiableTopLevelItemGroup;
+import org.jenkinsci.plugins.workflow.job.WorkflowJob;
+
+import javax.annotation.CheckForNull;
+import javax.annotation.Nonnull;
+import java.io.UnsupportedEncodingException;
+import java.net.URLDecoder;
+import java.net.URLEncoder;
+
+/**
+ * @author Vivek Pandey
+ */
+@Extension(ordinal = -9999)
+public class BlueOceanUrlMapperImpl extends BlueOceanUrlMapper {
+    @Override
+    public String getUrl(@Nonnull ModelObject modelObject) {
+        BlueOrganization organization = getOrganization(modelObject);
+        if(organization == null){ //no organization, best we can do is to land user on landing page
+            return getLandingPagePath();
```

Comment:
@alvarolobato `getLandingPagePath()` is computed using `BlueOceanUIProvider`, so should be fine, anyways ordinal value is going to let other plugins override it.